frontend/src/scenes/data-warehouse/editor/sidebar/EditorSidebar.tsx

```diff
@@ -32,12 +38,20 @@ export const EditorSidebar = ({
     }, [sidebarWidth])
 
     // State to track active tab
-    const [activeTab, setActiveTab] = useState('query_database')
+    const [activeTab, setActiveTab] = useState(EditorSidebarTab.QueryDatabase)
+    const prevLengthRef = useRef(0)
+
+    useEffect(() => {
+        if (variablesForInsight.length > prevLengthRef.current) {
+            setActiveTab(EditorSidebarTab.QueryVariables)
+        }
+        prevLengthRef.current = variablesForInsight.length
+    }, [variablesForInsight])
```

Comment:
Think this is fine, may have been nicer in a logic, but its simple enough to live here I reckon